dc/s2n-quic-dc/src/stream/server/tokio/tcp/worker.rs

```diff
@@ -226,25 +258,130 @@ enum WorkerState {
 }
 
 impl WorkerState {
-    fn poll<Sub, Pub>(
+    fn poll<Sub, Pub, B>(
         &mut self,
         cx: &mut task::Context,
-        context: &mut Context<Sub>,
+        context: &mut Context<Sub, B>,
         stream: &mut Option<(LazyBoundStream, SocketAddress)>,
         subscriber_ctx: &mut Option<Sub::ConnectionContext>,
         queue_time: Timestamp,
         now: Timestamp,
         publisher: &Pub,
+        unix_socket_path: &Option<String>,
```

Comment:
Paths should generally be `Path` or `PathBuf`, not String.

I'd also consider whether we should make workers store the type implementing PollBehavior and keep the per-behavior state (e.g., this path) in that implementation, rather than threading it through. That would eliminate the Option, at minimum, and makes it a bit cleaner overall I think.